api/src/user/dao/user.dao.ts

```diff
@@ -2,28 +2,33 @@ import { Injectable } from '@nestjs/common';
 import { v4 as uuidv4 } from 'uuid';
 import { PrismaService } from '../../prisma/prisma.service';
 import { CreateUserDto } from '../dto/create-user.dto';
+import { QueryDto } from '../dto/query.dto';
 import { FriendStatus } from '../user.service';
 import { Role } from '@prisma/client';
 
 @Injectable()
 export class UserDao {
   constructor(private readonly prisma: PrismaService) {}
 
-  async getUser(userId: string) {
-    return await this.prisma.user.findUniqueOrThrow({
-      where: { id: userId },
-    });
-  }
-
-  async getUserByUsername(username: string) {
-    return await this.prisma.user.findUniqueOrThrow({
-      where: { username },
+  async getUsers(query: QueryDto) {
+    return await this.prisma.user.findMany({
+      where: {
+        AND: [
+          {
+            id: query.id,
+            username: query.username,
+            email: query.email,
+          },
+        ],
+      },
+      take: query.page || 10,
```

Comment:
I think this is reversed, no? `page` to me should be `page_number` and `page_number` should be `perPage`?

`{{BASE_URL}}/users?page=3&page-number=1` returns 3 users on the first page.

Reply:
> {{BASE_URL}}/users?page=3&page-number=1 returns 3 users on the first page.

Ins't this how it works with my changes?